### DISCUSSION TITLE
subclassing FastAPI

### DISCUSSION THREAD

**Discussion by TobiasHT5** [fastapi/fastapi#6868] [Questions] [answered]
I tried recently to subclass a FastAPI class to create a more concrete class. I didn't get any python errors but the response from FastAPI was the normal JSON error response. And all examples I've seen, I've not seen anywhere were FastAPI is subclassed so far.
So is it possible to subclass FastAPI class?

**Comment by ycd** [comment]
It is not possible to see where you made mistakes without any code, Can you follow the issue template?

**Comment by TobiasHT5** [comment]
```python
from fastapi import FastAPI

class NewAPI(FastAPI):
      def __init__(self):
           super(NewAPI,self).__init__()
      def get(self,path:str = "/home"):
           return {"greeting":"welcome home"}
```
that's sort of an example of what i was talking about. the one i wrote didn't have any python errors. but instead of returning the response of NewAPI's get method, it instead return an error JSON file
``` json
{
    "detail": [
        {
            "loc": [
                "path",
                "item_id"
            ],
            "msg": "value is not a valid integer",
            "type": "type_error.integer"
        }
    ]
}
```

**Comment by Mause** [comment, marked as answer]
That error does not appear to be related to the code you have provided.

And to answer your question, no, the `FastAPI` class is not intended to be subclassed to implement routes.

**Comment by ycd** [comment]
> ```python
> from fastapi import FastAPI
> 
> class NewAPI(FastAPI):
>       def __init__(self):
>            super(NewAPI,self).__init__()
>       def get(self,path:str = "/home"):
>            return {"greeting":"welcome home"}
> ```
> 

Ahhh, i see, use class-based views from dmontagu's fastapi-utils.

It will be way more cleaner solution than subclassing FastAPI.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue @ht-thomas  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Comment by mdevino** [comment]
Hello. I'm subclassed FastAPI to override the default exception handlers. Is that okay?
It seems to work for me, I'm just wondering if there are any gotchas, or if I should do it some other way.

Context: I'm writing a API that will implement common functionality and models, so others who want to create a similar API can get these functionalities out of the box, and only worry about their own API logic.
